structured_type_test.go

```diff
@@ -1621,3 +1623,135 @@ func forAllStructureTypeFormats(dbt *DBTest, f func(t *testing.T, format string)
+			var rows driver.Rows
+			err = dbt.conn.Raw(func(sc any) error {
+				queryer, implementsQueryContext := sc.(driver.QueryerContext)
+				assertTrueF(t, implementsQueryContext, "snowflake conn does not implement QueryerContext but needs to")
+				rows, err = queryer.QueryContext(ctx, query, nil)
+				return err
+			})
+			assertNilF(t, err, fmt.Sprintf("test failed to run the following query: %s", query))
+			defer rows.Close()
+
+			sfRows, isSfRows := rows.(SnowflakeRows)
+			assertTrueF(t, isSfRows, "rows are not snowflakeRows")
+
+			batches, err := sfRows.GetArrowBatches()
+			assertNilF(t, err)
+			assertNotEqualF(t, len(batches), 0)
+			batch, err := batches[0].Fetch()
+			assertNilF(t, err)
+			assertNotEqualF(t, len(*batch), 0)
+			for _, record := range *batch {
+				// check number of cols/rows so we dont get index out of range
+				assertEqualF(t, record.NumRows(), int64(1), "wrong number of rows")
+				assertEqualF(t, record.NumRows(), int64(1), "wrong number of cols")
+
+				colIndex := 0
+				rowIndex := 0
+
+				srtCol, isStrCol := record.Column(colIndex).(*array.String)
+				assertTrueF(t, isStrCol, "column is not type string")
+
+				assertEqualE(t, srtCol.Value(rowIndex), "")
+				isNull := srtCol.IsNull(rowIndex)
+				assertTrueF(t, isNull)
+				record.Release()
+			}
+		}
+	})
+}
+
+func TestStructuredTypeInArrowBatchesWithSemistructuredObject(t *testing.T) {
+	skipStructuredTypesTestsOnGHActions(t)
+	runDBTest(t, func(dbt *DBTest) {
+		objectQueries := []string{
+			"SELECT 1, {'s':'someString'}::OBJECT(s VARCHAR)", // returns string
+			"SELECT {'s':'someString'}::OBJECT",               // returns binary
+		}
+		withUtf8Validation := []bool{
+			true,
+			false,
+		}
+		for i := 0; i < len(objectQueries); i++ {
+			for j := 0; j < len(withUtf8Validation); j++ {
+				shouldUtf8Verify := withUtf8Validation[i]
+				query := objectQueries[i]
+
+				pool := memory.NewCheckedAllocator(memory.DefaultAllocator)
+				defer pool.AssertSize(t, 0)
+				ctx := WithArrowBatches(WithArrowAllocator(context.Background(), pool))
+				if shouldUtf8Verify {
+					ctx = WithArrowBatchesUtf8Validation(ctx)
+				}
+
+				var err error
+				var rows driver.Rows
+				err = dbt.conn.Raw(func(sc any) error {
+					queryer, implementsQueryContext := sc.(driver.QueryerContext)
+					assertTrueF(t, implementsQueryContext, "snowflake conn does not implement QueryerContext but needs to")
+					rows, err = queryer.QueryContext(ctx, query, nil)
+					return err
+				})
+				assertNilF(t, err)
+				defer rows.Close()
+
+				sfRows, isSfRows := rows.(SnowflakeRows)
+				assertTrueF(t, isSfRows, "rows are not snowflakeRows")
+
+				batches, err := sfRows.GetArrowBatches()
+				assertNilF(t, err)
+				assertNotEqualF(t, len(batches), 0)
+				batch, err := batches[0].Fetch()
+				assertNilF(t, err)
+				assertNotEqualF(t, len(*batch), 0)
+				for _, record := range *batch {
+					// if we are running the first query there should be two columns, otherwise one
```

Comment:
First column can be safely removed from query, the test should simplify then.

Reply:
Done, thanks